src/set_theory/zfc/basic.lean

```diff
@@ -857,6 +866,17 @@ theorem map_unique {f : Set.{u} → Set.{u}} [H : definable 1 f] {x z : Set.{u}}
 λ h, ⟨λ y yx, let ⟨z, zx, ze⟩ := mem_image.1 yx in ze ▸ pair_mem_prod.2 ⟨zx, h z zx⟩,
      λ z, map_unique⟩⟩
 
+/-- Given a predicate `p` on ZFC sets. `hereditarily p x` means that `x` has property `p` and the
+members of `x` are all `hereditarily p`. -/
+def hereditarily (p : Set → Prop) (x : Set) : Prop :=
```

Comment:
Is it true that `hereditarily p x → p Ø`? That might be a nice lemma to prove as well.

Reply:
```lean
lemma empty_of_hereditarily {p : Set → Prop} {x : Set} : hereditarily p x → p ∅ :=
begin
  apply x.induction_on,
  intros y IH h,
  rcases Set.eq_empty_or_nonempty y with (rfl|⟨a, ha⟩),
  { exact h.def.1 },
  { exact IH a ha (h.def.2 _ ha) }
end
```